src/themes/default/components/core/Columns.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <div class="columns">
+    <select
+      name="column"
+      class="cl-secondary"
+      v-model="column"
+      @change="changeColumn">
+      <option v-for="(option) in productsColumns" :value="option" :key="option">{{ option }}</option>
+    </select>
+  </div>
+</template>
+
+<script>
+export default {
+  data () {
+    return {
+      column: 3,
+      productsColumns: this.$store.state.config.products.columns
+    }
+  },
+  methods: {
+    changeColumn () {
+      this.$bus.$emit('column-change', this.column)
```

Comment:
Ideally we shouldn't use event bus in any of the new components since it's gonna be depreciated. Please use props/component events instead https://vuejs.org/v2/guide/components-props.html#One-Way-Data-Flow